In [1]:
import numpy as np
import pandas as pd
from pandas import Series,DataFrame
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score,confusion_matrix


In [2]:
features=pd.read_csv('vaccinePredFeatures.csv')
Labels=pd.read_csv('vaccinePredLabels.csv')


In [3]:
Consolidated = features.merge(Labels, on='respondent_id')
Consolidated

respondent_id  h1n1_concern  h1n1_knowledge  behavioral_antiviral_meds  \
0                  0           1.0             0.0                        0.0   
1                  1           3.0             2.0                        0.0   
2                  2           1.0             1.0                        0.0   
3                  3           1.0             1.0                        0.0   
4                  4           2.0             1.0                        0.0   
...              ...           ...             ...                        ...   
26702          26702           2.0             0.0                        0.0   
26703          26703           1.0             2.0                        0.0   
26704          26704           2.0             2.0                        0.0   
26705          26705           1.0             1.0                        0.0   
26706          26706           0.0             0.0                        0.0   

       behavioral_avoidance  behavioral_face_mask  behavioral_wash_hands  \
0                       0.0                   0.0                    0.0   
1                       1.0                   0.0                    1.0   
2                       1.0                   0.0                    0.0   
3                       1.0                   0.0                    1.0   
4                       1.0                   0.0                    1.0   
...                     ...                   ...                    ...   
26702                   1.0                   0.0                    0.0   
26703                   1.0                   0.0                    1.0   
26704                   1.0                   1.0                    1.0   
26705                   0.0                   0.0                    0.0   
26706                   1.0                   0.0                    0.0   

       behavioral_large_gatherings  behavioral_outside_home  \
0                              0.0                      1.0   
1                              0.0                      1.0   
2                              0.0                      0.0   
3                              1.0                      0.0   
4                              1.0                      0.0   
...                            ...                      ...   
26702                          0.0                      1.0   
26703                          0.0                      0.0   
26704                          1.0                      0.0   
26705                          0.0                      0.0   
26706                          0.0                      0.0   

       behavioral_touch_face  ...  rent_or_own   employment_status  \
0                        1.0  ...          Own  Not in Labor Force   
1                        1.0  ...         Rent            Employed   
2                        0.0  ...          Own            Employed   
3                        0.0  ...         Rent  Not in Labor Force   
4                        1.0  ...          Own            Employed   
...                      ...  ...          ...                 ...   
26702                    0.0  ...          Own  Not in Labor Force   
26703                    0.0  ...         Rent            Employed   
26704                    1.0  ...          Own                 NaN   
26705                    NaN  ...         Rent            Employed   
26706                    0.0  ...          Own  Not in Labor Force   

       hhs_geo_region                census_msa  household_adults  \
0            oxchjgsf                   Non-MSA               0.0   
1            bhuqouqj  MSA, Not Principle  City               0.0   
2            qufhixun  MSA, Not Principle  City               2.0   
3            lrircsnp       MSA, Principle City               0.0   
4            qufhixun  MSA, Not Principle  City               1.0   
...               ...                       ...               ...   
26702        qufhixun                   

In [4]:
Consolidated.to_csv("VPoutput.csv", index=False)

In [5]:
Consolidated.shape


(26707, 38)

In [6]:
Consolidated.isnull().sum()


respondent_id                      0
h1n1_concern                      92
h1n1_knowledge                   116
behavioral_antiviral_meds         71
behavioral_avoidance             208
behavioral_face_mask              19
behavioral_wash_hands             42
behavioral_large_gatherings       87
behavioral_outside_home           82
behavioral_touch_face            128
doctor_recc_h1n1                2160
doctor_recc_seasonal            2160
chronic_med_condition            971
child_under_6_months             820
health_worker                    804
health_insurance               12274
opinion_h1n1_vacc_effective      391
opinion_h1n1_risk                388
opinion_h1n1_sick_from_vacc      395
opinion_seas_vacc_effective      462
opinion_seas_risk                514
opinion_seas_sick_from_vacc      537
age_group                          0
education                       1407
race                               0
sex                                0
income_poverty                  4423
m

In [7]:
consolidatedCleanedFile=Consolidated.dropna()
consolidatedCleanedFile.to_csv("VPcleanedFile.csv", index=False)
consolidatedCleanedFile.isnull().sum()

respondent_id                  0
h1n1_concern                   0
h1n1_knowledge                 0
behavioral_antiviral_meds      0
behavioral_avoidance           0
behavioral_face_mask           0
behavioral_wash_hands          0
behavioral_large_gatherings    0
behavioral_outside_home        0
behavioral_touch_face          0
doctor_recc_h1n1               0
doctor_recc_seasonal           0
chronic_med_condition          0
child_under_6_months           0
health_worker                  0
health_insurance               0
opinion_h1n1_vacc_effective    0
opinion_h1n1_risk              0
opinion_h1n1_sick_from_vacc    0
opinion_seas_vacc_effective    0
opinion_seas_risk              0
opinion_seas_sick_from_vacc    0
age_group                      0
education                      0
race                           0
sex                            0
income_poverty                 0
marital_status                 0
rent_or_own                    0
employment_status              0
hhs_geo_re

In [8]:
consolidatedCleanedFile.shape

(6437, 38)

#Define X and y

In [9]:
X = consolidatedCleanedFile.loc[:,['h1n1_concern','behavioral_antiviral_meds','behavioral_avoidance','behavioral_face_mask','behavioral_wash_hands','behavioral_large_gatherings','behavioral_touch_face','h1n1_knowledge', 'opinion_h1n1_vacc_effective','opinion_h1n1_risk','opinion_h1n1_sick_from_vacc','doctor_recc_h1n1']]
y = consolidatedCleanedFile.h1n1_vaccine

#train test and split 

In [10]:
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.4, random_state=13)

In [11]:
X_train.shape

(3862, 12)

In [12]:
X_test.shape

(2575, 12)

In [13]:
model = LogisticRegression()
model.fit(X_train,y_train) #Training the model in  machine learning
y_predict = model.predict(X_test)
accuracy_score(y_test,y_predict)

0.8054368932038835

In [14]:
y_test

24114    0
1204     1
11414    0
8784     0
18006    0
        ..
1483     0
20784    0
1121     1
10828    0
9181     1
Name: h1n1_vaccine, Length: 2575, dtype: int64

In [15]:
y_predict

array([0, 1, 0, ..., 1, 0, 0], dtype=int64)

#Accuracy of the model that predicts the persons likely to receive H1N1 Vaccine is 80.5%

In [16]:
pd.crosstab(y_test,y_predict) #o -No 1 yes

col_0            0    1
h1n1_vaccine           
0             1629  184
1              317  445

#Defining u and V

In [17]:
U = consolidatedCleanedFile.loc[:,['behavioral_large_gatherings','behavioral_touch_face','behavioral_antiviral_meds','behavioral_avoidance','behavioral_face_mask','behavioral_wash_hands','opinion_seas_vacc_effective', 'opinion_seas_risk','opinion_seas_sick_from_vacc','doctor_recc_seasonal']]
v = consolidatedCleanedFile.seasonal_vaccine

In [18]:
U_train, U_test, v_train, v_test = train_test_split(U,v,test_size=0.4, random_state=13)

In [19]:
U_train.shape

(3862, 10)

In [20]:
U_test.shape

(2575, 10)

In [21]:
model = LogisticRegression() #Initializing the model 
model.fit(U_train,v_train) #Training the model in  machine learning
v_predict = model.predict(U_test) #prediction of the target variable v using features U
accuracy_score(v_test,v_predict)

0.7704854368932039

#Accuracy of the model that predicts the people those are likely to receive seasonal Vaccine is 77%

In [22]:
pd.crosstab(v_test,v_predict) #o -No 1 yes

col_0                0    1
seasonal_vaccine           
0                 1125  260
1                  331  859